### PR TITLE
Plfm 4247 d

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/ViewSnapshotDao.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package org.sagebionetworks.repo.model.dbo.dao.table;
+
+import org.sagebionetworks.repo.model.entity.IdAndVersion;
+
+public interface ViewSnapshotDao {
+
+	/**
+	 * Create a record of a view snapshot stored in S3.
+	 * 
+	 * @param userId
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/ViewSnapshotDao.java]: params

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/ViewSnapshotDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		dbo.setCreatedBy(dto.getCreatedBy());
+		dbo.setCreatedOn(new Timestamp(dto.getCreatedOn().getTime()));
+		dbo.setBucket(dto.getBucket());
+		dbo.setKey(dto.getKey());
+		return dbo;
+	}
+
+	static ViewSnapshot translate(DBOViewSnapshot in) {
+		return new ViewSnapshot().withSnapshotId(in.getSnapshotId()).withViewId(in.getViewId())
+				.withVersion(in.getVersion()).withCreatedBy(in.getCreatedBy()).withCreatedOn(in.getCreatedOn())
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/ViewSnapshotDaoImpl.java]: create a date

File: lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/table/ViewSnapshotDaoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	@Autowired
+	private ViewSnapshotDao viewSnapshotDao;
+
+	ViewSnapshot viewSnapshot;
+	IdAndVersion idAndVersion;
+	long adminUserId;
+
+	@BeforeEach
+	public void beforeEach() {
+		adminUserId = BOOTSTRAP_PRINCIPAL.THE_ADMIN_USER.getPrincipalId();
```
Comment (john-hill) [inline lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/table/ViewSnapshotDaoImplTest.java:33]: truncate before

File: lib/lib-utils/src/main/java/org/sagebionetworks/util/FileProvider.java [hunk 4]
```diff
@@ -28,6 +30,16 @@
...
+	
+	/**
+	 * Create a Writer for the given file.
+	 * @param file
+	 * @param charsetName
+	 * @return
+	 * @throws FileNotFoundException
+	 * @throws UnsupportedEncodingException 
+	 */
+	Writer createFileWriter(File file, String charsetName) throws FileNotFoundException, UnsupportedEncodingException;
```
Comment (john-hill) [inline lib/lib-utils/src/main/java/org/sagebionetworks/util/FileProvider.java]: User standard charset that does not throw Unsupported.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/BucketAndKey.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+package org.sagebionetworks.repo.manager.table;
+
+/**
+ * Buck and Key pair.
+ *
+ */
+public class BucketAndKey {
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/BucketAndKey.java]: move to models

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableViewManager.java [hunk 6]
```diff
@@ -90,6 +91,7 @@ public void updateEntityInView(UserInfo user,
 	 * @param tableId
 	 * @param snapshotOptions
 	 * @return
+	 * @throws IOException
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableViewManager.java]: cleanup exception

**Review by marcomarasca — APPROVED** [review]

**Pull request merged**